# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieCommons.h
# Commit: d01a1a055fec
# Full Hash: d01a1a055fece12b906b938954f984bd69278ef7
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-04-30 20:49:53
# Regressor Bug: 1960824
# File Overlap Count: 1
# Description:
#   Bug 1960824 - Introduce nsICookie::SAMESITE_UNSET, r=edgul,webdriver-reviewers,extension-reviewers,cookie-reviewers,valentin,Sasha,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246638
# ==============================================================================

diff -r 307ff391dd02 -r d01a1a055fec netwerk/cookie/CookieCommons.h
--- a/netwerk/cookie/CookieCommons.h	Wed Apr 30 08:53:12 2025 +0000
+++ b/netwerk/cookie/CookieCommons.h	Wed Apr 30 08:59:18 2025 +0000
@@ -115,15 +115,16 @@
   static already_AddRefed<nsICookieJarSettings> GetCookieJarSettings(
       nsIChannel* aChannel);
 
-  static bool ShouldIncludeCrossSiteCookie(Cookie* aCookie,
+  static bool ShouldIncludeCrossSiteCookie(Cookie* aCookie, nsIURI* aHostURI,
                                            bool aPartitionForeign,
                                            bool aInPrivateBrowsing,
                                            bool aUsingStorageAccess,
                                            bool aOn3pcbException);
 
   static bool ShouldIncludeCrossSiteCookie(
-      int32_t aSameSiteAttr, bool aCookiePartitioned, bool aPartitionForeign,
-      bool aInPrivateBrowsing, bool aUsingStorageAccess, bool aOn3pcbException);
+      nsIURI* aHostURI, int32_t aSameSiteAttr, bool aCookiePartitioned,
+      bool aPartitionForeign, bool aInPrivateBrowsing, bool aUsingStorageAccess,
+      bool aOn3pcbException);
 
   static bool IsFirstPartyPartitionedCookieWithoutCHIPS(
       Cookie* aCookie, const nsACString& aBaseDomain,